packages/nuxt/src/pages/runtime/plugins/router.ts

```diff
@@ -261,9 +261,9 @@ const plugin: Plugin<{ router: Router }> = defineNuxtPlugin({
       await nuxtApp.callHook('page:loading:end')
     })
 
-    router.afterEach(async (to, _from) => {
+    router.afterEach((to) => {
       if (to.matched.length === 0) {
-        await nuxtApp.runWithContext(() => showError(createError({
+        nuxtApp.runWithContext(() => showError(createError({
```

Comment:
```suggestion
        return nuxtApp.runWithContext(() => showError(createError({
```

Reply:
Thanks for the suggestion! I'm curious, why do we need `return` here?

Reply:
I'm not sure whether vue router awaits this function or not. by returning the promise we can avoid the async/await within our code, and leave it up to vue router whether to await it